torch/csrc/jit/api/module_save.cpp

```diff
@@ -5,11 +5,30 @@ namespace torch {
 namespace jit {
 
 void Module::save(std::ostream& out, const ExtraFilesMap& extra_files) const {
+#ifdef FBCODE_CAFFE2
+  if (this->type()->getForwardHooks().size() > 0 || this->type()->getForwardPreHooks().size() > 0) {
+    throw std::runtime_error(
+          "Cannot save module '" + this->type()->name()->name() 
+          + "' because it has forward hooks or pre-hooks attached. " +
+          "Saving modules with hooks not supported in FBCODE yet. Please " +
+          "remove the hooks before scripting if you want to save this model.");
+
+  }
+#endif
```

Comment:
I would have put this in the serialization PR because that's what it's related to but this entire stack is landing together so I think it should be okay.